### PR TITLE
Added data validation for registration form. Closed #5

### LINKED ISSUE #5: Add data validation for registration form

### PR DESCRIPTION
**Pull request by London3149-train**
Added data validation for registration form. Closed #5

### CODE REVIEW

**Review by SugarSalt9 — DISMISSED** [review]

**Review by SugarSalt9 — DISMISSED** [review]

**Review by Andrey31ts — DISMISSED** [review]

File: messenger-client/Services/Interfeces/IBlockInjections.cs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+﻿namespace messenger_client.Services.Interfeces
+{
+    public interface IBlockInjections
+    {
+        public bool CheckForProhibitedSymbol(string str);
```
Comment (TopAcademy2022) [inline messenger-client/Services/Interfeces/IBlockInjections.cs]: Remove "For" from method name. Rename method parameter.

File: messenger-client/Pages/Registration.cshtml.cs [hunk 2]
```diff
@@ -1,12 +1,65 @@
...
-        public void OnGet()
-        {
-        }
-    }
+	public class RegistrationModel : PageModel
+	{
+		private IBlockInjections blockInjections = new BlockInjections();
+		public string IncorrectDataMessage { get; set; }
+
+		public void OnGet(bool regRes)
```
Comment (TopAcademy2022) [inline messenger-client/Pages/Registration.cshtml.cs]: Delete method parameter and clear context. Or remove method.

File: messenger-client/Pages/Registration.cshtml.cs [hunk 3]
```diff
@@ -1,12 +1,65 @@
...
 {
-    public class RegistrationModel : PageModel
-    {
-        public void OnGet()
-        {
-        }
-    }
+	public class RegistrationModel : PageModel
+	{
+		private IBlockInjections blockInjections = new BlockInjections();
```
Comment (TopAcademy2022) [inline messenger-client/Pages/Registration.cshtml.cs:10]: Add space.

**Review by xs887z — DISMISSED** [review]

File: messenger-client/Pages/Registration.cshtml.cs [hunk 4]
```diff
@@ -1,12 +1,65 @@
...
+
+		public IActionResult OnPostRegistration([FromBody] RegistrationData data)
+		{
+			IncorrectDataMessage = String.Empty;
+
+			string? login = data.Login;
+			string? password = data.Password;
+			string? email = data.Email;
+
+			if (login != String.Empty && password != String.Empty && email != String.Empty)
```
Comment (TopAcademy2022) [inline messenger-client/Pages/Registration.cshtml.cs]: Replace this. Using String.IsNullOrEmpty() method.

File: messenger-client/Services/BlockInjections.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
+﻿using messenger_client.Services.Interfeces;
+using static System.Net.Mime.MediaTypeNames;
+
+namespace messenger_client.Services
+{
+    public class BlockInjections : IBlockInjections
+    {
+        private char[] _ListProhibitedSymbol;
```
Comment (TopAcademy2022) [inline messenger-client/Services/BlockInjections.cs]: Add space.

File: messenger-client/Services/BlockInjections.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
+﻿using messenger_client.Services.Interfeces;
+using static System.Net.Mime.MediaTypeNames;
```
Comment (TopAcademy2022) [inline messenger-client/Services/BlockInjections.cs]: What the library?

File: messenger-client/Services/BlockInjections.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+namespace messenger_client.Services
+{
+    public class BlockInjections : IBlockInjections
+    {
+        private char[] _ListProhibitedSymbol;
+        public BlockInjections()
+        {
+            this._ListProhibitedSymbol = new char[] { '\'', '\"', '=', '*', '#', '|', '$','\\' };
+		}
+        public bool CheckForProhibitedSymbol(string verifiedText)
```
Comment (TopAcademy2022) [inline messenger-client/Services/BlockInjections.cs]: Add space.

Comment (TopAcademy2022) [reply messenger-client/Services/BlockInjections.cs]: Replace method with LINQ.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

File: messenger-client/Services/BlockInjections.cs [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+	public class BlockInjections : IBlockInjections
+	{
+		private char[] _ListProhibitedSymbol;
+
+		public BlockInjections()
+		{
+			this._ListProhibitedSymbol = new char[] { '\'', '\"', '=', '*', '#', '|', '$', '\\' };
+		}
+
+		public bool CheckProhibitedSymbols(string verifiedText)
```
Comment (TopAcademy2022) [inline messenger-client/Services/BlockInjections.cs:14]: Replace method with LINQ.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

**Review by xs887z — DISMISSED** [review]

**Review by Andrey31ts — DISMISSED** [review]

**Review by SugarSalt9 — DISMISSED** [review]

File: messenger-client/Services/BlockInjections.cs [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+		{
+			bool rezult = true;
```
Comment (TopAcademy2022) [inline messenger-client/Services/BlockInjections.cs]: Optimize this, remove variable.

**Review by TopAcademy2022 — CHANGES_REQUESTED** [review]

**Review by TopAcademy2022 — DISMISSED** [review]

**Review by TopAcademy2022 — APPROVED** [review]

**Review by xs887z — APPROVED** [review]

**Review by SugarSalt9 — APPROVED** [review]

**Review by Mi-stik31 — APPROVED** [review]

**Pull request merged**